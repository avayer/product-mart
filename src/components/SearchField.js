import React from 'react';

class SearchField extends React.Component {

    onSearchSubmit = (e) => {
        this.props.onTermSubmit(e.target.value);
    }

    render() {
        return(
                <input type="text" placeholder="Search products" onChange={e => this.onSearchSubmit(e)} />
        );
    }
} 

export default SearchField;