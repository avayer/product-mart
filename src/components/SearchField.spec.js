import React from "react";
import { shallow } from "enzyme";

import SearchField from './SearchField';

describe("Testing SearchField Component", () => {
  it("Snapshot of SearchField component", () => {
    const wrapper = shallow(<SearchField />);

    expect(wrapper).toMatchSnapshot();
  });

  it("check for input field", () => {
    const wrapper = shallow(<SearchField />);

    expect(wrapper.containsMatchingElement(<input/>)).toEqual(true);
  });
});
