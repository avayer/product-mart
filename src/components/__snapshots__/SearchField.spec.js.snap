// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SearchField Component Snapshot of SearchField component 1`] = `
<div
  className="ui one column stackable center aligned page grid"
  style={
    Object {
      "marginTop": "5px",
    }
  }
>
  <div
    class="ui action input"
  >
    <input
      onChange={[Function]}
      placeholder="Search products"
      type="text"
    />
    <button
      className="ui icon button"
      disabled={true}
    >
      <i
        class="search icon"
      />
    </button>
  </div>
</div>
`;
